### ISSUE TITLE
Still calling to cloud on local install

### ISSUE THREAD

**Issue by veioenza** [macarthur-lab/seqr#1424]
I performed a local installation using the guide at https://github.com/macarthur-lab/seqr/blob/master/deploy/LOCAL_INSTALL.md
The end goal is for this to run on an on-prem cluster but for now it is just running on a single node. When I try to run the example at the end of the documentation I get the following error.

python3 -m seqr_loading SeqrMTToESTask --local-scheduler \
    --reference-ht-path /seqr-reference-data/GRCh38/combined_reference_data_grch38.ht \
    --clinvar-ht-path /seqr-reference-data/GRCh38/clinvar.GRCh38.2020-06-15.ht \
    --vep-config-json-path /vep-GRCh38.json \
    --es-host elasticsearch \
    --es-index-min-num-shards 3 \
    --sample-type WES \
    --es-index pidtrio \
    --genome-version 38 \
    --source-paths /vep_data/PID_trio.vcf \
    --dest-path /vep_data/PID_trio.mt 

Welcome to
     __  __     <>__
    / /_/ /__  __/ /
   / __  / _ `/ / /
  /_/ /_/\_,_/_/_/   version 0.2.39-ef87446bd1c7
LOGGING: writing to /seqr-reference-data/GRCh38/hail-20200814-1158-0.2.39-ef87446bd1c7.log
ERROR: [pid 1328] Worker Worker(salt=633754082, workers=1, host=a16e165f9914, username=root, pid=1328) failed    SeqrVCFToMTTask(source_paths=/vep_data/PID_trio.vcf, dest_path=/vep_data/PID_trio.mt, genome_version=38, vep_runner=VEP, reference_ht_path=/seqr-reference-data/GRCh38/combined_reference_data_grch38.ht, clinvar_ht_path=/seqr-reference-data/GRCh38/clinvar.GRCh38.2020-06-15.ht, hgmd_ht_path=None, sample_type=WES, dont_validate=False, dataset_type=VARIANTS, remap_path=, subset_path=, vep_config_json_path=/vep-GRCh38.json)
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/luigi/worker.py", line 199, in run
    new_deps = self._run_get_new_deps()
  File "/usr/local/lib/python3.7/site-packages/luigi/worker.py", line 141, in _run_get_new_deps
    task_gen = self.task.run()
  File "/hail-elasticsearch-pipelines/luigi_pipeline/seqr_loading.py", line 55, in run
    self.read_vcf_write_mt()
  File "/hail-elasticsearch-pipelines/luigi_pipeline/seqr_loading.py", line 64, in read_vcf_write_mt
    self.validate_mt(mt, self.genome_version, self.sample_type)
  File "/hail-elasticsearch-pipelines/luigi_pipeline/seqr_loading.py", line 107, in validate_mt
    sample_type_stats = HailMatrixTableTask.sample_type_stats(mt, genome_version)
  File "/hail-elasticsearch-pipelines/luigi_pipeline/lib/hail_tasks.py", line 99, in sample_type_stats
    ht = hl.read_table(ht_path)
  File "<decorator-gen-1296>", line 2, in read_table
  File "/usr/local/lib/python3.7/site-packages/hail/typecheck/check.py", line 585, in wrapper
    return __original_func(*args_, **kwargs_)
  File "/usr/local/lib/python3.7/site-packages/hail/methods/impex.py", line 2348, in read_table
    for rg_config in Env.backend().load_references_from_dataset(path):
  File "/usr/local/lib/python3.7/site-packages/hail/backend/spark_backend.py", line 324, in load_references_from_dataset
    return json.loads(Env.hail().variant.ReferenceGenome.fromHailDataset(self.fs._jfs, path))
  File "/usr/local/lib/python3.7/site-packages/py4j/java_gateway.py", line 1257, in __call__
    answer, self.gateway_client, self.target_id, self.name)
  File "/usr/local/lib/python3.7/site-packages/hail/backend/spark_backend.py", line 38, in deco
    'Error summary: %s' % (deepest, full, hail.__version__, deepest)) from None
hail.utils.java.FatalError: HailException: MatrixTable and Table files are directories; path 'gs://seqr-reference-data/GRCh38/validate_ht/common_noncoding_variants.grch38.ht' is not a directory

Java stack trace:
is.hail.utils.HailException: MatrixTable and Table files are directories; path 'gs://seqr-reference-data/GRCh38/validate_ht/common_noncoding_variants.grch38.ht' is not a directory
        at is.hail.utils.ErrorHandling$class.fatal(ErrorHandling.scala:9)
        at is.hail.utils.package$.fatal(package.scala:76)
        at is.hail.expr.ir.RelationalSpec$.readMetadata(AbstractMatrixTableSpec.scala:32)
        at is.hail.expr.ir.RelationalSpec$.readReferences(AbstractMatrixTableSpec.scala:66)
        at is.hail.variant.ReferenceGenome$.fromHailDataset(ReferenceGenome.scala:587)
        at is.hail.variant.ReferenceGenome.fromHailDataset(ReferenceGenome.scala)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
        at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
        at py4j.Gateway.invoke(Gateway.java:282)
        at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
        at py4j.commands.CallCommand.execute(CallCommand.java:79)
        at py4j.GatewayConnection.run(GatewayConnection.java:238)
        at java.lang.Thread.run(Thread.java:748)

Hail version: 0.2.39-ef87446bd1c7
Error summary: HailException: MatrixTable and Table files are directories; path 'gs://seqr-reference-data/GRCh38/validate_ht/common_noncoding_variants.grch38.ht' is not a directory
DEBUG: 1 running tasks, waiting for next task to finish
INFO: Informed scheduler that task   SeqrVCFToMTTask__seqr_reference__VARIANTS__vep_data_PID_tr_bda8d2f4ef   has status   FAILED
DEBUG: Asking scheduler for work...
DEBUG: Done
DEBUG: There are no more tasks to run at this time
DEBUG: There are 2 pending tasks possibly being run by other workers
DEBUG: There are 2 pending tasks unique to this worker
DEBUG: There are 2 pending tasks last scheduled by this worker
INFO: Worker Worker(salt=633754082, workers=1, host=a16e165f9914, username=root, pid=1328) was stopped. Shutting down Keep-Alive thread
INFO:
===== Luigi Execution Summary =====

Scheduled 3 tasks of which:
* 1 complete ones were encountered:
    - 1 VcfFile(filename=/vep_data/PID_trio.vcf)
* 1 failed:
    - 1 SeqrVCFToMTTask(...)
* 1 were left pending, among these:
    * 1 had failed dependencies:
        - 1 SeqrMTToESTask(...)

This progress looks :( because there were failed tasks

===== Luigi Execution Summary =====

**Comment by veioenza** [comment]
This seems to be solved by using the "--SeqrMTToESTask-dont-validate" flag, but is there no method to download and point to a local validation set?

**Comment by bw2** [comment]
Glad you found a work-around. The 4 validation file paths are in
https://github.com/macarthur-lab/hail-elasticsearch-pipelines/blob/master/luigi_pipeline/lib/global_config.py 
and I haven't tested this, but after you download those, you may be able to switch to local paths by updating your luigi config as in https://github.com/macarthur-lab/hail-elasticsearch-pipelines/blob/master/luigi_pipeline/configs/seqr-loading-local-GRCh38.cfg#L2-L3

**Comment by veioenza** [comment]
Thanks for your help, I'll give that a go on the next run. So the non-validated run finished fine with no errors but when I try to use the es-index value to load the data I get.

Variant call dataset path must end with .vcf.gz or .vds or .bed

Should the source file always been gzipped from the start? Using gzip on the source doesn't help now so do I need to do it all again with the vcf file gzipped?

**Comment by hanars** [comment]
You can modify the index metadata to avoid this error: 

- Go to the "staff pages" link on the top of the page next to the search bar
- Select "kibana" 
- select "dev tools" on the left side menu
- run the command `GET <es-index-name>/_mapping/variant`
- copy the "_meta" section of the output and modify the `sourceFilePath` to end with `.vcf.gz`
- update the index meta using the modified "_meta" value by running:
```
PUT  <es-index-name>/_mapping/variant
{"_meta": { 
    "genomeVersion": ...,
    "sampleType": ...,
    ...
    "sourceFilePath": "<your_file>.vcf.gz"   
  }
```

**Comment by veioenza** [comment]
Unfortunately that didn't help, in the end I had to bgzip the test file and re-add. Just for future information the local installation doesn't mention starting kibana so I needed do that first,
`docker-compose up -d kibana `

Regarding the validation data, again the work around didn't help but I did the following to get the cloud pull to work,
`docker-compose exec pipeline-runner python3 -m pip install git+https://github.com/bw2/hail-utils.git`

Either way I am now able to process a vcf.gz file with Hail. This file has 3 samples in it so I need to use a mapping file. I added it to vep_data, the same location as the source but it seems seqr can't find this. Looking in the seqr container I can't see any mount for this directory. Where should I put the mapping file to load it locally?
